src/js/relayclient/RelayClient.js

```diff
@@ -650,6 +577,92 @@ class RelayClient {
 
     return relayHub
   }
+
+  async _prepareRelayHttpRequest (
+    encodedFunction,
+    relayAddress,
+    pctRelayFee,
+    baseRelayFee,
+    gasPrice,
+    gasLimit,
+    paymaster,
+    relayHub,
+    options) {
+    const senderNonce = (await relayHub.methods.getNonce(options.to, options.from).call()).toString()
+    const relayRequest = new RelayRequest({
+      senderAddress: options.from,
+      target: options.to,
+      encodedFunction,
+      senderNonce: senderNonce.toString(),
+      pctRelayFee: pctRelayFee.toString(),
+      baseRelayFee: baseRelayFee.toString(),
+      gasPrice: gasPrice.toString(),
+      gasLimit: gasLimit.toString(),
+      paymaster,
+      relayAddress
+    })
+
+    const signature = await this._prepareSignature(relayHub, relayRequest, options)
+    let approvalData = options.approvalData || '0x'
+    if (typeof options.approveFunction === 'function') {
+      approvalData = '0x' + await options.approveFunction({
+        from: options.from,
+        to: options.to,
+        encodedFunctionCall: encodedFunction,
+        pctRelayFee: options.pctRelayFee,
+        gas_price: gasPrice,
+        gas_limit: gasLimit,
+        nonce: senderNonce,
+        relay_hub_address: relayHub._address,
+        relay_address: relayAddress
+      })
+    }
+    // max nonce is not signed, as contracts cannot access addresses' nonces.
+    let allowedRelayNonceGap = this.config.allowed_relay_nonce_gap
+    if (typeof allowedRelayNonceGap === 'undefined') {
+      allowedRelayNonceGap = 3
+    }
+    const relayMaxNonce = (await this.web3.eth.getTransactionCount(relayAddress)) + allowedRelayNonceGap
+    return { relayRequest, relayMaxNonce, approvalData, signature }
+  }
+
+  async _prepareSignature (relayHub, relayRequest, options) {
+    if (this.web3.eth.getChainId === undefined) {
```

Comment:
deosn't belong here. should be called early (e.g. at the constructor)

Reply:
I literally only extracted code that was in relayTransacrtion as-is to a function. That's it.